src/main/java/com/vmware/vipclient/i18n/util/LocaleUtility.java

```diff
@@ -19,7 +19,8 @@
     
     // A locale fallback priority queue. For now, it only contains the default locale and the source messages (from "messages_source.json").
     private static List<Locale> fallbackLocales = new LinkedList<Locale>(Arrays.asList(getDefaultLocale(), Locale.forLanguageTag(ConstantsKeys.SOURCE)));
-    
+    private static List<Locale> l2FallbackLocales = new LinkedList<Locale>(Arrays.asList(getDefaultLocale(), Locale.ENGLISH));
```

Comment:
I still disagree in having 2 separate fallback queue for L2 and L3. I think it creates unnecessary complexity of having to maintain 2 separate lists, 2 sets of setters, etc. Let's analyze the difference and similarities of L2 and L3 fallback behavior first. @gong-yu mentioned that we should have a discussion after 0.5.0 is released, so I created an issue here: https://github.com/vmware/singleton/issues/843

Questions:
1.  
Why does L2 need English in addition to the default locale that was set? Is it because the default locale may not be available while the English locale is always available? If yes, why? My concern is that we are tryin to remove any hardcoded locale from the code, but here you just hardcoded English.
2. What is the expected fallback behavior for L2? Isn't supposed to fallback to the default locale only, or should it also fallback to thew source locale (I do not see issue 758 anymore)? I  thought source locale is only meant for L3?

Reply:
We have discussed this item in last Friday's sprint review meeting, I have updated the conclusion to issue #758 , so I updated the code according to the conclusion, sorry for forgetting to let you know.

1. Yes, but I think we should consider correct behavior first, then code wise. Now the conclusion is the last fallback is English, so I have to hardcode it in code.

2. The conclusion of the correct fallback behavior for L2 is:
requested locale -> default locale -> English locale

Reply:
Just talked to Demin, no need to hardcode English.